server/controller/apis/goalController.js

```diff
@@ -169,6 +169,7 @@ exports.saveGoal = async ( req, res, redirect ) => {
         if( existingGoal ) {
             console.log( "there was an existing goal for this id: " + JSON.stringify( existingGoal ) );
             goal = existingGoal;
+            goal.goalVersion++;
```

Comment:
Looks like the new functionality will increment the version number even if the existing goal is saved with no changes. If this is intentional due to thinking that use case will be very rare, I don't mind approving. However, it looks like checking for changes would be as simple as comparing the existingGoal fields with the goal passed in via the req parameter.